ketcher-autotests/tests/File-Management/Open-And-Save-Files/Rxn-Files/rxn-files.spec.ts

```diff
@@ -1,12 +1,278 @@
+   * Description: Reaction with atom and bond properties
+   */
+  const xOffsetFromCenter = 40;
+  await page.goto('');
+  await drawBenzeneRing(page);
+  await selectNestedTool(page, BondTool.SINGLE_AROMATIC);
+  await clickOnBond(page, BondType.DOUBLE, 1);
+  await selectNestedTool(page, BondTool.TRIPPLE);
+  await clickOnBond(page, BondType.DOUBLE, 1);
+  await selectAtomInToolbar(AtomButton.Nitrogen, page);
+  await clickOnAtom(page, 'C', 1);
+  await selectNestedTool(page, ArrowTool.ARROW_FILLED_BOW);
+  await clickOnTheCanvas(page, xOffsetFromCenter, 0);
+
+  const expectedFile = await getRxn(page);
+  await saveToFile(
+    'RXN/reaction-with-atom-and-bond-properties-saved.rxn',
+    expectedFile,
+  );
+
+  await selectAction(TopPanelButton.Clear, page);
+
+  await openFileAndAddToCanvas(
+    'RXN/reaction-with-atom-and-bond-properties-saved.rxn',
+    page,
+  );
+
+  await takeEditorScreenshot(page);
+});
+
+test('Open and Save file - Reaction from file that contains Rgroup', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1901
+   * Description: Reaction from file that contains Rgroup
+   */
+  const xOffsetFromCenter = 40;
+  await page.goto('');
+  await drawBenzeneRing(page);
+  await selectLeftPanelButton(LeftPanelButton.R_GroupLabelTool, page);
+  await clickOnAtom(page, 'C', 1);
+  await page.getByRole('button', { name: 'R7' }).click();
+  await page.getByRole('button', { name: 'Apply' }).click();
+  await selectNestedTool(page, ArrowTool.ARROW_FILLED_BOW);
+  await clickOnTheCanvas(page, xOffsetFromCenter, 0);
+  await selectTopPanelButton(TopPanelButton.Save, page);
+  const saveButtonOne = page.getByRole('button', { name: 'Save', exact: true });
+  await expect(saveButtonOne).not.toHaveAttribute('disabled', 'disabled');
+
+  await pressButton(page, 'Cancel');
+  await selectNestedTool(page, RgroupTool.ATTACHMENT_POINTS);
+  await setAttachmentPoints(
+    page,
+    { label: 'C', index: 2 },
+    { primary: true },
+    'Apply',
+  );
+  await selectTopPanelButton(TopPanelButton.Save, page);
+  const saveButtonTwo = page.getByRole('button', { name: 'Save', exact: true });
+  await expect(saveButtonTwo).not.toHaveAttribute('disabled', 'disabled');
+
+  await page.getByRole('button', { name: 'Cancel' }).click();
+  await selectNestedTool(page, RgroupTool.R_GROUP_FRAGMENT);
+  const { x, y } = await getCoordinatesTopAtomOfBenzeneRing(page);
+  await page.mouse.click(x, y);
+  await page.getByRole('button', { name: 'R22' }).click();
+  await page.getByRole('button', { name: 'Apply' }).click();
+  await selectTopPanelButton(TopPanelButton.Save, page);
+  const saveButtonThree = page.getByRole('button', {
+    name: 'Save',
+    exact: true,
+  });
+  await expect(saveButtonThree).not.toHaveAttribute('disabled', 'disabled');
+});
+
+// EPMLSOPKET-1903 - TO DO
+// TO DO: while saving file showed ERROR: array: invalid index -2 (size=0)
+// Need to create new bug looks like it connected to #2389 issue
```

Comment:
did you discuss it with @Zhirnoff ?

Reply:
Yes, he will check this and create a bug